typescript/web/src/hooks/__tests__/use-images-navigation.tsx

```diff
@@ -1,90 +1,58 @@
+/* eslint-disable import/first */
 import { renderHook } from "@testing-library/react-hooks";
-import { useRouter } from "next/router";
-import { incrementMockedDate } from "../../../../dev-utils/mockdate";
-import { client } from "../../connectors/apollo-client/schema-client";
-import { processImage } from "../../connectors/repository/image-processing";
-import { setupTestsWithLocalDatabase } from "../../utils/setup-local-db-tests";
+import { mockNextRouter } from "../../utils/router-mocks";
+import { BASIC_DATASET_DATA } from "../../utils/tests/data.fixtures";
+import { CURRENT_IMAGE } from "../test-utils";
+
+mockNextRouter({
+  isReady: true,
+  query: {
+    imageId: CURRENT_IMAGE.id,
+    datasetSlug: BASIC_DATASET_DATA.slug,
+    workspaceSlug: BASIC_DATASET_DATA.workspace.slug,
+  },
+});
 import {
-  CREATE_TEST_DATASET_MUTATION,
-  CREATE_TEST_IMAGE_MUTATION,
-} from "../../utils/tests/mutations";
-import { useImagesNavigation } from "../use-images-navigation";
-
-setupTestsWithLocalDatabase();
-
-jest.mock("../../connectors/repository/image-processing");
-const mockedProcessImage = processImage as jest.Mock;
+  CURRENT_NOT_IN_IMAGES_MOCKS,
+  IMAGE1,
+  IMAGE2,
+  IMAGE_IS_FIRST_MOCKS,
+  IMAGE_IS_LAST_MOCKS,
+  NO_IMAGES_MOCKS,
+  THREE_IMAGES_MOCKS,
+} from "../use-image-navigation.fixtures";
 
-const testDatasetId = "mocked-dataset-id";
-
-jest.mock("next/router", () => ({
-  useRouter: jest.fn(() => ({
-    query: { datasetSlug: "test-dataset", workspaceSlug: "local" },
-  })),
-}));
+import { useImagesNavigation } from "../use-images-navigation";
+import {
+  getApolloMockLink,
+  getApolloMockWrapper,
+} from "../../utils/tests/apollo-mock";
 
-async function createImage(name: String) {
-  mockedProcessImage.mockReturnValue({
-    width: 42,
-    height: 36,
-    mime: "image/jpeg",
-  });
-  const mutationResult = await client.mutate({
-    mutation: CREATE_TEST_IMAGE_MUTATION,
-    variables: {
-      file: new Blob(),
-      name,
-      datasetId: testDatasetId,
-    },
-  });
+const NoImagesWrapper = getApolloMockWrapper(
+  getApolloMockLink(NO_IMAGES_MOCKS)
+);
 
-  const {
-    data: {
-      createImage: { id },
-    },
-  } = mutationResult;
-
-  return id;
-}
-
-beforeEach(async () => {
-  await client.mutate({
-    mutation: CREATE_TEST_DATASET_MUTATION,
-    variables: {
-      name: "test dataset",
-      datasetId: testDatasetId,
-      workspaceSlug: "local",
-    },
-  });
-});
+const ThreeImagesWrapper = getApolloMockWrapper(
+  getApolloMockLink(THREE_IMAGES_MOCKS)
+);
 
-const Wrapper = ({ children }: React.PropsWithChildren<{}>) => (
-  <ApolloProvider client={client}>{children}</ApolloProvider>
+const ImageNotInDatasetWrapper = getApolloMockWrapper(
+  getApolloMockLink(CURRENT_NOT_IN_IMAGES_MOCKS)
 );
 
 test("The currentImageIndex is undefined while loading", async () => {
   const { result } = renderHook(() => useImagesNavigation(), {
-    wrapper: Wrapper,
+    wrapper: NoImagesWrapper,
```

Comment:
Instead of declaring custom wrappers for each variant, you could factorize a little bit more and have a ` renderTest` function looking like this one:

```tsx
const renderTest = (mocks: ApolloMockResponses) => {
  const link = getApolloMockLink(mocks);
  const wrapper = getApolloMockWrapper(link);
  return renderHook(() => useImagesNavigation(), { wrapper });
};
```

Then, in each test, just call something like:

```ts
const { result, waitForValueToChange } = renderTest(CURRENT_NOT_IN_IMAGES_MOCKS);
```